docs/flamework/additional-modules/components/configuration.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Configuration
+---
+Components have a few features we haven't covered yet!
+
+## Predicate
+You can add a `predicate` function to a component to determine if the component should be constructed for a provided instance.
+
+This only affects Components created via **CollectionService**!
+```ts
+@Component({
+    tag: "Example",
+    predicate: ( instance ) => instance.FindFirstAncestorOfClass("PlayerGui") !== undefined,
+})
+export class ExampleComponent extends BaseComponent<> implements OnStart {}
+```
+
+## Instance Guard
+The `instanceGuard` is similar to `predicate`, but serves a slightly different purpose.
+
+* On the server, an error will be thrown if a component is attempted to be created, but the `instanceGuard` fails.
+* On the client, a connection will be created to wait for `DescendantsAdded` & `DescendantsRemoving` until the `instanceGuard` passes. This is because instances can be sent to the client in pieces and the whole thing might not have arrived yet.
+```ts
+interface Guard extends Tool {
+    Parent: Workspace
+}
+
+const instanceGuard = Flamework.createGuard<Guard>()
+
+@Component( {
+    tag: "Example",
+    instanceGuard: ( instance ): instance is Guard => instanceGuard(instance) && instance.Parent === Workspace,
+} )
+export class ExampleComponent extends BaseComponent<> implements OnStart {}
+```
+
+## Defaults
+You can use `defaults` to specify the value of the Component's attributes if they were not set before creating the Component.
+
+Normally, you must assign all the attributes to a Component otherwise it will not be created. If you specify the `defaults` though, the Component can be created without specifying those attributes.
+```ts
+interface Attributes {
+    amount: number
+}
+
+@Component( {
+    tag: "Example",
+    defaults: {
+        "amount": 1
+    }
+} )
+export class ExampleComponent extends BaseComponent<Attributes, Part> implements OnStart {}
+```
+
+## Refresh Attributes
+(Default `true`)
+
+Specifies if the Component should update its `attributes` when `SetAttribute` is called on the Component after it has been created.
+
+If set to `false`, the Components attributes (`this.attributes.example`) will not be updated to the value set when using `SetAttribute`, after the Component has been created.
```

Comment:
It should be noted that this also disables the `onAttributeChanged` handlers.

That might be changed in the future but it's worth documenting for now.